apiml-common/src/main/java/com/ca/mfaas/product/gateway/GatewayInstanceInitializer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.product.gateway;
+
+import com.ca.mfaas.product.constants.CoreService;
+import com.ca.mfaas.product.instance.InstanceInitializationException;
+import com.ca.mfaas.product.instance.lookup.InstanceLookupExecutor;
+import com.netflix.appinfo.InstanceInfo;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+
+import java.net.URI;
+
+@Slf4j
+@RequiredArgsConstructor
+public class GatewayInstanceInitializer {
+
+    private final InstanceLookupExecutor instanceLookupExecutor;
+    private final ApplicationEventPublisher applicationEventPublisher;
+    private final GatewayClient gatewayClient;
+
+    private GatewayConfigProperties process(InstanceInfo instanceInfo) {
+        try {
+            String gatewayHomePage = instanceInfo.getHomePageUrl();
+            URI uri = new URI(gatewayHomePage);
+
+            return GatewayConfigProperties.builder()
+                .scheme(uri.getScheme())
+                .hostname(uri.getHost() + ":" + uri.getPort())
+                .build();
+        } catch (Exception e) {
+            throw new InstanceInitializationException(e.getMessage());
+        }
+
+    }
+
+    @EventListener
+    public void init(ApplicationReadyEvent applicationReadyEvent) {
```

Comment:
`applicationReadyEvent` never used...

Reply:
Event parameter is mandatory for event listener method by Spring. So it's the parameter to detect whether the spring context is ready.
But I found an alternative way of detecting. I will change. Thanks